### DISCUSSION TITLE
HalfmoonUI Playground

### DISCUSSION THREAD

**Discussion by Supernova3339** [halfmoonui/halfmoon#137] [Show and tell]
# Introducing HalfmoonUI Playground

![HalfmoonUI Playground](https://www.gethalfmoon.com/static/images/halfmoon-icon.svg)

I'm excited to introduce the **HalfmoonUI Playground**, an interactive online environment where you can experiment with HalfmoonUI and create stunning user interfaces! [Source Code](https://github.com/Supernova3339/halfmoonui-playground)

## Features

- Code Completion for Halfmoon CSS
- Clearing Code for a Fresh Start
- Sharing Code with Others
- ( NEW ) Theme Switcher

## Why Use the Playground?

The HalfmoonUI Playground is perfect for:

- Learning how the Halfmoon CSS framework works
- Prototyping new ideas and UI components
- Creating demos to share online

## How It Works

1. Edit the HalfmoonUI code on the left side.
2. See the live preview on the right side.
3. Use the "Share" button to copy your code to the clipboard and share it with others.

## Get Started

1. Visit the HalfmoonUI Playground: [HalfmoonUI Playground](https://supernova3339.github.io/halfmoonui-playground/)
2. Try out the provided sample code or write your own.
3. Experiment, create, and enjoy the power of HalfmoonUI!

### Code View
<img width="705" alt="image" src="https://github.com/halfmoonui/halfmoon/assets/63515814/a130a353-4e5e-43eb-9afb-208a9409035c">

### Output Preview
<img width="717" alt="image" src="https://github.com/halfmoonui/halfmoon/assets/63515814/b048bc5c-3519-4f1b-a09c-0a8c9eff3de5">

## Ready to Dive In?

Join the fun at the [HalfmoonUI Playground](https://supernova3339.github.io/halfmoonui-playground/) and start creating beautiful HalfmoonUI designs today!

Feel free to leave your feedback, questions, or share your creations in the comments below. Happy coding! 🌙

**Comment by Supernova3339** [comment]
If you would like to use this on the HalfmoonUI website, please create a separate repository so we can keep the source code categorized.
Cheers, SuperDev 🎉 

P.S: I am working on the snippets extension 👀

**Comment by HarshNarayanJha** [comment]
Wow, it's a great idea. And now even useful since to migrate from v1 to v2 we will have to prototype things out now and then

**Reply by Supernova3339** [reply]
Thanks for your feedback!

**Comment by Supernova3339** [comment]
To make HalfmoonUI Playground easier to use, we have added a domain. For easier access to the site, please use the following domain:
[halfmoonplayground.oo.gd](https://halfmoonplayground.oo.gd)

**Comment by FlavioSuarez** [comment]
Great work!! Thank you!

**Reply by Supernova3339** [reply]
Thanks for your feedback!

**Comment by Supernova3339** [comment]
Hey everyone :wave:

HalfmoonUI 2.0.1 has released! I'm working on the version and style switcher right now.

A week or less to release!

@halfmoonui should I have the latest version available by default in the selector?

Once I've got this squashed, I might rewrite everything with NextJS so we can have more awesomeness :tada:

The offers always open to change it to playground.gethalfmoon.com :)

**Comment by Supernova3339** [comment]
Might look into puterjs to have the share function deploy your code online directly!
